File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
stack = []

word = input()
bomb_word = input()
bomb_size = len(bomb_word)

size = 0

for c in word:
    if size >= bomb_size:
        if ''.join(stack[-bomb_size:]) == bomb_word:
            for _ in range(bomb_size):
                stack.pop()
                size -= 1

    stack.append(c)
    size += 1

if size >= bomb_size:
    if ''.join(stack[-bomb_size:]) == bomb_word:
        for _ in range(bomb_size):
            stack.pop()
            size -= 1

if stack:
    print("".join(stack))
else:
    print("FRULA")

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <bits/stdc++.h>
using namespace std;

stack<char> stk;

string word, bomb, result;
int prev_idx = 0;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	cin >> word;
	cin >> bomb;

	for (char c : word) {
		stk.push(c);

		if (stk.size() >= bomb.size()) {
			string temp = "";
			for (int i = 0; i < bomb.size(); ++i) {
				temp += stk.top();
				stk.pop();
			}
			reverse(temp.begin(), temp.end());

			if (temp != bomb) {
				for (char c : temp)
					stk.push(c);
			}
		}
	}

	if (stk.size()) {
		while (stk.size()) {
			result += stk.top(); stk.pop();
		}
		reverse(result.begin(), result.end());
		cout << result;
	}
	else {
		cout << "FRULA";
	}

	return 0;
}